Tribler/Core/Upgrade/db72_to_pony.py

```diff
@@ -134,89 +132,88 @@ def get_old_torrents_count(self, personal_channel_only=False):
-                                      (" == " if personal_channel_only else " != ") + \
-                                      (" %i " % self.personal_channel_id)
+        with contextlib.closing(sqlite3.connect(self.tribler_db)) as connection, connection:
+            cursor = connection.cursor()
+            cursor.execute("PRAGMA temp_store = 2")
+
+            personal_channel_filter = ""
+            if self.personal_channel_id:
+                personal_channel_filter = " AND ct.channel_id " + \
+                                          (" == " if personal_channel_only else " != ") + \
+                                          (" %i " % self.personal_channel_id)
+
+            torrents = []
+            batch_not_empty = False # This is a dumb way to indicate that this batch got zero entries from DB
+
+            for tracker_url, channel_id, name, infohash, length, creation_date, torrent_id, category, num_seeders, \
+                num_leechers, last_tracker_check in \
+                    cursor.execute(
+                        self.select_full + personal_channel_filter + " group by infohash" +
+                        (" LIMIT " + str(batch_size) + " OFFSET " + str(offset))):
+                batch_not_empty = True
+                # check if name is valid unicode data
+                try:
+                    name = str(name)
+                except UnicodeDecodeError:
+                    continue
 
-        torrents = []
-        batch_not_empty = False # This is a dumb way to indicate that this batch got zero entries from DB
-
-        for tracker_url, channel_id, name, infohash, length, creation_date, torrent_id, category, num_seeders, \
-            num_leechers, last_tracker_check in \
-                cursor.execute(
-                    self.select_full + personal_channel_filter + " group by infohash" +
-                    (" LIMIT " + str(batch_size) + " OFFSET " + str(offset))):
-            batch_not_empty = True
-            # check if name is valid unicode data
-            try:
-                name = str(name)
-            except UnicodeDecodeError:
-                continue
-
-            try:
-                if (len(base64.decodestring(infohash.encode('utf-8'))) != 20) or \
-                        (not torrent_id or int(torrent_id) == 0) or \
-                        (not length or (int(length) <= 0) or (int(length) > (1 << 45))) or \
-                        (is_forbidden(name)) or \
-                        not name:
+                try:
+                    invalid_decoding = len(base64.decodestring(infohash.encode('utf-8'))) != 20
+                    invalid_id = not torrent_id or int(torrent_id) == 0
+                    invalid_length = not length or (int(length) <= 0) or (int(length) > (1 << 45))
+                    invalid_name = not name or is_forbidden(name)
+                    if invalid_decoding or invalid_id or invalid_length or invalid_name:
```

Comment:
Nice refactoring! This really makes the code more readable :+1:

Reply:
Thank you 😄